examples/example-three/netlify.yml

```diff
@@ -0,0 +1,13 @@
+build:
+  lifecycle:
+    init:
+      - uses: netlify-plugin-one
+      - uses: netlify-plugin-two#doThis
+    build:
+      - echo 'Hello Netlify Build!' && npm run makeSite
+      - uses: netlify-plugin-three
+      - uses: netlify-plugin-two#doThat
+        config:
+          # Plugin-to-user communication
+          # Uses the `zaz` return value of `netlify-plugin-three`
+          fizz: ${outputs:zaz}
```

Comment:
How does this know where zaz is from?

Reply:
This is returned by `netlify-plugin-one`. Outputs is initially an empty object, filled in by each lifecycle command/plugin.

Reply:
Does this mean outputs from `init` can not be used in a `preDeploy` phase?

Reply:
Outputs can be re-used by any later lifecycle (or later command in the same lifecycle).
So outputs from `init` would be usable in a `preDeploy` phase.

Reply:
If I have an output `{ zaz: 'hithere'}` from a plugin in `init` and then a plugin in `preBuild` with an output `{ zaz: 'goodbye'}`. Is the output from `init` plugin overwritten by the `preBuild` output of the same keyspace?

Reply:
Yes. This enables:
  - plugins explicitly meant to override another plugin
  - plugins to communicate "I am consuming 'foo' property" while allowing any other plugins to set that value, without knowing those plugins. This decouples outputs consumption from production: plugins can know each other, but they don't have to.

That being said, if a plugin wants to prevent other plugins from overriding its output, it can also namespace the output:

```js
return { myPluginName: { zaz: true } }
```

Reply:
yeah outputs most likely will need to be namespaced to avoid this. 

The mutability is interesting but sounds like a hard thing for users to debug

Reply:
Sure, we can support both or either of those.